Add End-User docs for Nextcloud (#108)
vshn/appcat-user-docs: docs/modules/ROOT/pages/vshn-managed/nextcloud/create.adoc

= Create a Nextcloud service

Apply the following object on your namespace, as specified by its YAML description.

.Example to create a Nextcloud instance
[source,yaml]
----
apiVersion: vshn.appcat.vshn.io/v1
kind: VSHNNextcloud
metadata:
  name: nextcloud-app1-prod # <1>
  namespace: prod-app # <2>
spec:
  parameters:
    service:
      version: "23" # <3>

>       defaultInternalDB: true # <4>

Reviewer: The name of this field is confusing. If it contains `default` I would assume this is on by default, but it's false by default.

Maybe `useInternalDB`?

Reviewer 2: We had some issues with the naming and came up with `defaultInternalDB`. I guess your point is valid and can go with your suggestion.

Author: Changed it to useExternalPostgreSQL

>       defaultInternalDB: true # <4>

Reviewer 2: we need to rethink this part that's very important!

>       defaultInternalDB: true # <4>

Reviewer 2: maybe we should use `managedPostgreSQL: true/false` and then `pgSettings`? Since we are pushing our PostgreSQL managed instance by default, there is not need to show the default DB. We can add that to the description of `managedPostgreSQL` parameter

>       defaultInternalDB: true # <4>

Reviewer 2: we agreed to use this name `useExternalPostgreSQL` variable and also reuse the [postgreSQLParameters](https://docs.appcat.ch/vshn-managed/keycloak/create.html)

      postgreSQLParameters: {} # <5>
    size: # <6>
      plan: standard-2
  writeConnectionSecretToRef:
    name: nextcloud-creds # <7>
----
<1> Instance name
<2> The namespace where the object will be created
<3> Nextcloud version

> <4> If we should use the VSHNPostgreSQL as a database backend instead of the build-in SqLite database backend. (defaults to true)

Reviewer 2: ```suggestion
<4> If we should use the VSHNPostgreSQL as a managed backend database instead of the build-in SQLite database. (defaults to true)
```

<5> PostgreSQL database parameters. See xref:vshn-managed/postgresql/index.adoc[PostgreSQL by VSHN] (Only used when `useExternalPostgreSQL` is false)
<6> Size of the Nextcloud instance. See xref:vshn-managed/nextcloud/plans.adoc[Plans and Sizing] for more information.

> <7> Secret where the connection details are provisioned. This secret shouldn't exist before creation.

Reviewer 2: ```suggestion
<7> Secret where the connection details are provisioned. This secret is created after the instance is provisioned.
```

Author: No. The secret shouldn't exist before the creation or crossplane will not write anything as an "unmanaged" sercret already exists.

Reviewer 2: Ah now I see from what point of view you are referring to. Fine by me.
